graphql/connection_test.go

```diff
@@ -65,31 +65,20 @@ func TestConnection(t *testing.T) {
 	item := schema.Object("item", Item{})
 	item.Key("id")
 	inner.FieldFunc("innerConnection", func(args Args) []Item {
-		retList := make([]Item, 5)
-		retList[0] = Item{Id: 1}
-		retList[1] = Item{Id: 2}
-		retList[2] = Item{Id: 3}
-		retList[3] = Item{Id: 4}
-		retList[4] = Item{Id: 5}
-		return retList
+		return []Item{{1}, {2}, {3}, {4}, {5}}
 	}, schemabuilder.Paginated)
+	inner.FieldFunc("innerConnectionWithFilter", func() []Item {
+		return []Item{{1}, {2}, {3}, {4}, {5}}
+	}, schemabuilder.Paginated, schemabuilder.TextFilterFields{
+		"foo": func(ctx context.Context, i Item) string {
```

Comment:
Do we have to define this for all filterable fields?  Or is there a way to point to a field on the item direclty?  Is this to allow filtering on multiple fields at the same time?

Reply:
Also, it looks like "foo" is not actually used anywhere in the query.  Is that just the name of the field?  Does it have value?  Can we reference it from the client?

Reply:
Ok, this is admittedly not great API right now - my thoughts here are that we want to add `filterBy` (where only certain fields are filtered) and this allows for that. Other than that, there's currently no use / no way the client can reference it. Currently the default would be to filter using all defined fields.